ui/src/app/machines/views/MachineList/FilterAccordion/FilterAccordion.js

```diff
@@ -0,0 +1,202 @@
+    return `${Math.round(
+      speedInMbytes / megabytesInGigabyte / gigabytesInTerabyte
+    )} Tbps`;
+  }
+
+  return `${speedInMbytes} Mbps`;
+};
+
+const getFilters = machines => {
+  const filters = new Map();
+  machines.forEach(machine => {
+    filterOrder.forEach(filter => {
+      let value = getMachineValue(machine, filter);
+      // This is not a useful value so skip it.
+      if (!value) {
+        return;
+      }
+      // Turn everything into an array so we can loop over all values.
+      if (!Array.isArray(value)) {
+        value = [value];
+      }
+      let storedFilter = filters.get(filter);
+      if (!storedFilter) {
+        filters.set(filter, new Map());
+        storedFilter = filters.get(filter);
+      }
+      value.forEach(filterValue => {
+        let storedValue = storedFilter.get(filterValue);
+        if (!storedValue) {
+          storedFilter.set(filterValue, 0);
+          storedValue = storedFilter.get(filterValue);
+        }
+        storedFilter.set(filterValue, storedValue + 1);
+      });
+    });
+  });
+  return filters;
+};
+
+const sortByFilterName = (a, b) => {
+  a = a[0];
+  b = b[0];
+  if (a < b) {
+    return -1;
+  }
+  if (a > b) {
+    return 1;
+  }
+  return 0;
+};
+
+const FilterAccordion = ({ searchText, setSearchText }) => {
+  const machines = useSelector(machineSelectors.all);
+  const machinesLoaded = useSelector(machineSelectors.loaded);
+  const filterOptions = useMemo(() => getFilters(machines), [machines]);
+  const currentFilters = getCurrentFilters(searchText);
+  let sections;
+  if (machinesLoaded) {
+    sections = filterOrder.reduce((options, filter) => {
+      const filterValues = filterOptions.get(filter);
+      if (filterValues) {
+        options.push({
+          title: filterNames.get(filter),
+          content: (
+            <List
+              className="u-no-margin--bottom"
+              items={Array.from(filterValues)
+                .sort(sortByFilterName)
+                .map(([filterValue, count]) => (
+                  <Button
+                    appearance="base"
+                    className={classNames(
+                      "u-align-text--left u-no-margin--bottom filter-accordion__item",
+                      {
+                        "is-active": isFilterActive(
+                          currentFilters,
+                          filter,
+                          filterValue,
+                          true
+                        )
+                      }
+                    )}
+                    onClick={() => {
+                      const newFilters = toggleFilter(
+                        currentFilters,
+                        filter,
+                        filterValue,
+                        true
+                      );
+                      setSearchText(filtersToString(newFilters));
+                    }}
+                  >
+                    {filter === "link_speeds"
```

Comment:
This is a special case for the display of the value, it needs to use the unchanged value in all other cases so that the search/filtering works correctly.